### PR TITLE
Diligen Details Implementation

### PR DESCRIPTION
**Pull request by TayGov**
# Description

This PR includes the following proposed change(s):

Implement get details service

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Does the change impact or break the Docker build?

- [ ] Yes
- [x] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [ ] No

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by alexjoybc — COMMENTED** [review]
Can you add the service implementation here:

https://github.com/bcgov/jag-file-submission/blob/6e0918ba9019ca4b7d4c9dd8f9061afe307eca17/src/backend/efiling-reviewer-api/src/main/java/ca/bc/gov/open/jag/efilingreviewerapi/document/DocumentsApiDelegateImpl.java#L75-L80

thanks

File: src/backend/efiling-reviewer-api/src/main/java/ca/bc/gov/open/jag/efilingreviewerapi/document/DocumentsApiDelegateImpl.java [hunk 1]
```diff
@@ -75,6 +75,9 @@ public DocumentsApiDelegateImpl(DiligenService diligenService, ExtractRequestMap
     public ResponseEntity<Void> documentEvent(UUID xTransactionId, DocumentEvent documentEvent) {
 
         logger.info("document {} status has changed to {}", documentEvent.getDocumentId(), documentEvent.getStatus());
+        //We won't do anything with this for now
+        Object result = diligenService.getDocumentDetails(documentEvent.getDocumentId());
```
Comment (alexjoybc) [inline src/backend/efiling-reviewer-api/src/main/java/ca/bc/gov/open/jag/efilingreviewerapi/document/DocumentsApiDelegateImpl.java:79]:
```suggestion
        diligenService.getDocumentDetails(documentEvent.getDocumentId());
```

File: src/backend/libs/efiling-diligen-client-starter/src/main/java/ca/bc/gov/open/jag/efilingdiligenclientstarter/AutoConfiguration.java [hunk 2]
```diff
@@ -77,7 +77,7 @@ public RestTemplate restTemplate() {
     public DiligenService diligenService(DiligenAuthService diligenAuthService, RestTemplate restTemplate, ApiClient apiClient) {
         ObjectMapper objectMapper = new ObjectMapper();
         objectMapper.disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
-        return new DiligenServiceImpl(restTemplate, diligenProperties, diligenAuthService, objectMapper);
+        return new DiligenServiceImpl(restTemplate, diligenProperties, diligenAuthService, objectMapper, new DocumentsApi(apiClient));
```
Comment (alexjoybc) [inline src/backend/libs/efiling-diligen-client-starter/src/main/java/ca/bc/gov/open/jag/efilingdiligenclientstarter/AutoConfiguration.java:80]: Should we have a documentApi bean and inject it in the method parameters ?

File: src/backend/libs/efiling-diligen-client/src/main/java/ca/bc/gov/open/efilingdiligenclient/diligen/DiligenServiceImpl.java [hunk 3]
```diff
@@ -74,6 +81,28 @@ public BigDecimal postDocument(String documentType, MultipartFile file) {
...
 
+    @Override
+    public Object getDocumentDetails(BigDecimal documentId) {
+
+        logger.debug("getting document details from diligen");
+
+        documentsApi.getApiClient().setBearerToken(diligenAuthService.getDiligenJWT(diligenProperties.getUsername(), diligenProperties.getPassword()));
+
+        try {
+            InlineResponse2003 result = documentsApi.apiDocumentsFileIdDetailsGet(documentId.intValue());
```
Comment (alexjoybc) [inline src/backend/libs/efiling-diligen-client/src/main/java/ca/bc/gov/open/efilingdiligenclient/diligen/DiligenServiceImpl.java:92]: should we return this result ?
We will probably use the rest of the information for processing...

**Review by alexjoybc — CHANGES_REQUESTED** [review]
Couple of suggestions, let me know what you think

File: src/backend/libs/efiling-diligen-client/src/main/java/ca/bc/gov/open/efilingdiligenclient/diligen/DiligenServiceImpl.java (hunk 3 above)
Comment (TayGov) [reply src/backend/libs/efiling-diligen-client/src/main/java/ca/bc/gov/open/efilingdiligenclient/diligen/DiligenServiceImpl.java:96]: I mapped a new object for return

File: src/backend/efiling-reviewer-api/src/main/java/ca/bc/gov/open/jag/efilingreviewerapi/document/DocumentsApiDelegateImpl.java [hunk 4]
```diff
@@ -75,6 +75,9 @@ public DocumentsApiDelegateImpl(DiligenService diligenService, ExtractRequestMap
     public ResponseEntity<Void> documentEvent(UUID xTransactionId, DocumentEvent documentEvent) {
 
         logger.info("document {} status has changed to {}", documentEvent.getDocumentId(), documentEvent.getStatus());
+        //We won't do anything with this for now
```
Comment (akroon3r) [inline src/backend/efiling-reviewer-api/src/main/java/ca/bc/gov/open/jag/efilingreviewerapi/document/DocumentsApiDelegateImpl.java:78]: maybe mark this is a TODO for the ability to find it?

**Review by akroon3r — APPROVED** [review]

**Pull request merged**